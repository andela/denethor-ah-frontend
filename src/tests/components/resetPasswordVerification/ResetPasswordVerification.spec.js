import React from 'react';
import { shallow } from 'enzyme';
import { 
  ResetPasswordVerification
 } from '../../../components/resetPasswordVerification/ResetPasswordVerification';
import axios from '../../../utils/axiosConfig';

jest.mock('../../../utils/axiosConfig');

let wrapper;
let value;

beforeAll(() => {
  wrapper = shallow(<ResetPasswordVerification  />);
  value = 'foobar@bar.com';
})

test('Should render ResetPasswordVerification correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('Should set state on email change', () => {
  wrapper.find('input').at(0).simulate('change', { target: { value } });
  expect(wrapper.state('email')).toBe(value);
});

test('Should verify on email blur', () => {
  wrapper.find('input').at(0).simulate('blur', { target: { value: 'asdakldas' } });
  expect(wrapper.state('emailError')).toBe('Please enter a valid email address');

  wrapper.find('input').at(0).simulate('blur', { target: { value } });
  expect(wrapper.state('emailError')).toBe('');
});

test('Should handle submit event', async () => {
  const value = 'foobar@foo.com';
  const resetPasswordVerification = axios.post.mockImplementationOnce(() => Promise.resolve({
    email: value 
  }));
  const preventDefault = jest.fn();
  const wrapper = shallow(<ResetPasswordVerification />);

  wrapper.find('button').simulate('click', { preventDefault });
  expect(preventDefault).toHaveBeenCalled();

  wrapper.find('input').at(0).simulate('change', { target: { value } });
  wrapper.find('button').simulate('click', { preventDefault });
  expect(wrapper.state('email')).toBe(value);
  expect(preventDefault).toHaveBeenCalled();
  expect(resetPasswordVerification).toHaveBeenCalled();
});

test('Should throw an error when ', async () => {
  const value = 'foobar@foo.com';
  const resetPasswordVerification = axios.post.mockImplementationOnce(() => Promise.reject({
    response: { status: 404 }
  }));
  const preventDefault = jest.fn();
  const wrapper = shallow(<ResetPasswordVerification />);

  wrapper.find('button').simulate('click', { preventDefault });
  expect(preventDefault).toHaveBeenCalled();

  wrapper.find('input').at(0).simulate('change', { target: { value } });
  wrapper.find('button').simulate('click', { preventDefault });
  expect(wrapper.state('email')).toBe(value);
  expect(preventDefault).toHaveBeenCalled();
  expect(resetPasswordVerification).toHaveBeenCalled();
});
