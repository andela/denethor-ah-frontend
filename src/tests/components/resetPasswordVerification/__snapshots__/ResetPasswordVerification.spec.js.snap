// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ResetPasswordVerification correctly 1`] = `
<div
  className="flex modal-wrapper"
>
  <div
    className="modal-form-container"
  >
    <div
      className="modal-header"
    >
      <div
        className="logo-container"
      >
        <h3>
          Enter Email
        </h3>
      </div>
    </div>
    <div
      className="reset-password-form__container"
    >
      <form
        className="reset-password-form__form"
      >
        <div>
          <small
            className="password-error-message"
          >
            Â 
          </small>
          <input
            className=""
            name="password"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="email"
            required={true}
            type="text"
            value=""
          />
        </div>
        <button
          className="button button--primary"
          disabled={false}
          onClick={[Function]}
        >
          <span
            className="button__text--login undefined"
          >
            Submit
          </span>
          <span />
        </button>
      </form>
    </div>
  </div>
</div>
`;
